tests/testthat/test-linter_tags.R

```diff
@@ -33,8 +33,30 @@ test_that("default_linters and default tag match up", {
   expect_identical(tagged_default, names(default_linters))
 })
 
+test_that("warnings occur only for deprecated linters", {
+  expect_silent(linters_with_tags(tags = NULL))
+  num_deprecated_linters <- nrow(available_linters(tags = "deprecated", exclude_tags = NULL))
+  warns_expected <- num_deprecated_linters
+  expect_silent({
+    withCallingHandlers(
+      linters_with_tags(tags = "deprecated", exclude_tags = NULL),
+      warning = function(w) {
+        if (warns_expected >= 1L && grepl("was deprecated", conditionMessage(w))) {
+          warns_expected <<- warns_expected - 1L
+          invokeRestart("muffleWarning")
+        } else {
+          w
+        }
+      }
+    )
+    if (warns_expected != 0L) {
```

Comment:
I think `expect_identical(warns_expected, 0L)` is more testthat-ish... in which case we may want to take an increment (not decrement) approach, then failure would be more readably `expect_identical(n_deprecation_warnings, num_deprecated_linters)`

Reply:
I agree, but was afraid of the failure info "warns_expected is not equal to 0" being misleading.
Counting from 0 and up should fix this problem though.